litgpt/finetune/lora.py

```diff
@@ -208,6 +208,14 @@ def main(
     if fabric.device.type == "cuda":
         fabric.print(f"Memory used: {torch.cuda.max_memory_allocated() / 1e9:.02f} GB")
 
+    # Final evaluation
+    val_loss = validate(
+        fabric, model, val_dataloader, tokenizer, dataclasses.replace(eval, max_iters=len(val_dataloader)), data
```

Comment:
Why do you need the `.replace()` call?

Reply:
Because I want to run on the entire dataset, not just the `eval.max_iters` which for the final validation in my opinion is appropriate here.

Reply:
From a practical standpoint this makes total sense. It's basically the "true" (vs the approximated) validation set performance. I wonder if we can communicate that somehow without making it too confusing. Maybe just by modifying the docstring for eval.max_iters